From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ppug <ppug@users.noreply.github.com>
Date: Sun, 13 Dec 2020 18:30:58 +1100
Subject: [PATCH] Paper is gay


diff --git a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
index 0f7f1503d07bdff7ce7644dfe21e444adb52c7db..d71fbacb221da7e8583bb001987161ae6fdb478f 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
@@ -350,11 +350,7 @@ public final class CraftMagicNumbers implements UnsafeValues {
     public String getTimingsServerName() {
         return com.destroystokyo.paper.PaperConfig.timingsServerName;
     }
-
-    @Override
-    public com.destroystokyo.paper.util.VersionFetcher getVersionFetcher() {
-        return new com.destroystokyo.paper.PaperVersionFetcher();
-    }
+    
 
     @Override
     public boolean isSupportedApiVersion(String apiVersion) {
